server/chat-plugins/room-events.ts

```diff
@@ -136,18 +186,57 @@ export const commands: ChatCommands = {
 
 			if (!target) return this.errorReply("Usage: /roomevents view [event name]");
 			target = toID(target);
-			if (!room.events[target]) return this.errorReply(`There is no such event named '${target}'. Check spelling?`);
 
+			const event = room.events[getEventIDByNameOrAlias(target, room)];
+			if (!event) return this.errorReply(`There is no such event named '${target}'. Check spelling?`);
 			if (!this.runBroadcast()) return;
-			const buff = `<table border="1" cellspacing="0" cellpadding="3">${formatEvent(room.events[target])}</table>`;
+			const buff = `<table border="1" cellspacing="0" cellpadding="3">${formatEvent(event)}</table>`;
 			this.sendReply(`|raw|<div class="infobox-limited">${buff}</div>`);
 			if (!this.broadcasting && user.can('ban', null, room)) {
 				this.sendReplyBox(
-					Chat.html`<code>/roomevents add ${room.events[target].eventName} |` +
-					Chat.html`${room.events[target].date} | ${room.events[target].desc}</code>`
+					Chat.html`<code>/roomevents add ${event.eventName} |` +
+					Chat.html`${event.date} | ${event.desc}</code>`
 				);
 			}
 		},
+		alias: 'addalias',
+		addalias(target, room, user) {
+			if (!room.chatRoomData) return this.errorReply("This command is unavailable in temporary rooms.");
+			if (!this.can('ban', null, room)) return false;
+			const [alias, eventId] = target.split(target.includes('|') ? '|' : ',').map(argument => toID(argument));
+			if (!(alias && eventId)) {
+				return this.errorReply("Usage: /roomevents addalias [alias], [event name]. Aliases must contain at least one alphanumeric character.");
+			}
+			if (!room.events?.[eventId]) return this.errorReply(`There is no event named "${eventId}".`);
```

Comment:
I would make a separate error message if the room has no events listed; something like `There are currently no scheduled events.`

Reply:
Also I would change "named" to "titled" to keep consistent

Reply:
Pretty sure the code I added in b1b7533 does that

Reply:
turns out I didn't fix the named vs titled thing, thats done now